### ISSUE TITLE
Mender upgrades across L4T releases (and yocto branches) are not currently supported

### ISSUE THREAD

**Issue by dwalkes** [OE4T/meta-mender-community#14]
Partition layout changes on L4T releases have made this not trivial to support for older yocto branches.

Changes are coming with L4T version 32.6 which we may be able to leverage to upgrade from older branches, and no partition changes are planned from r32.5 (gatesgarth) forward at this time.

See OE4T meeting notes at [this link](https://docs.google.com/document/u/2/d/e/2PACX-1vShY0B20I5JDb00u1nR8fDbBJMiyXoh-TgY87LCHQCtCK7fvK0jAIVggNoEZhiG01ItSahUDBnv_TLY/pub) and watch meeting updates in [this link](https://github.com/OE4T/meta-tegra/discussions/515) for more information.

See [this sheet](https://docs.google.com/spreadsheets/d/14zVltS2LTwCNywoje6_8wzFzHQ3sYVhY6C7T0DkOnx0/edit#gid=2052166477) which tracks some combination of changes to partition layout between releases and has some references to associated xml files where these changes are tracked.

See [this commit](https://github.com/Trellis-Logic/meta-mender-community-old/pull/10/commits/7536c741444b4add6f48a84c97bf815b93142170) which backs out a feature to support u-boot environment partition migration between releases, which could be useful for TX2 u-boot based mender implementations.

**Comment by dwalkes** [comment]
See related thread at https://hub.mender.io/t/mender-standalone-update-from-warrior-to-zeus-or-dunfell/3516/8

**Comment by hoostol** [comment]
any updates on this issue?

**Comment by dwalkes** [comment]
Yes, there have been lots of discussions about this in Monthly meetings, see https://github.com/OE4T/meta-tegra/wiki/Over-the-air-reflashing-process and meeting notes starting from May at https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes---2021-05-13, especially https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2021-06-10

Most recent is probably https://github.com/OE4T/meta-tegra/discussions/798 and https://github.com/OE4T/tegra-demo-distro/pull/154/commits/6a582f6df6866ca1d27cc51217bdc33d51862ec9

Summary:  
* Updates across the most recent L4T > r32.4.3 releases should work seamlessly, but we still need help testing these and porting changes across branches.
* Updates from any version should be possible (we've demonstrated from JP 3.3 successfully) but require some work, outlined in https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2021-06-10 and https://github.com/OE4T/meta-tegra/wiki/Over-the-air-reflashing-process.

**Comment by dwalkes** [comment]
One other related thread about making a more general solution can be found here:
https://github.com/OE4T/tegra-demo-distro/discussions/117
